### PR TITLE
pdpb: support scan regions for range batch

### PR DESCRIPTION
**Pull request by you06**
ref pingcap/tidb#53850

Add an interface which batch scan regions for key ranges.

The test result:

1. split table into 1000 regions by `SPLIT TABLE sbtest1 BETWEEN (0) AND (1000000) REGIONS 1000;`
2. wait until region cache expired.
3. run a simple query `explain analyze select * from sbtest1 limit 1;`

- Without this interface: `...build_task_duration: 81.5ms...`, statmenet execution takes 0.11 sec
- With this interface: `...build_task_duration: 2.7ms...`, statement execution takes 0.02 sec

### CODE REVIEW

**Review by JmPotato — APPROVED** [review]

**Review by HuSharp — APPROVED** [review]

File: proto/pdpb.proto [hunk 1]
```diff
@@ -53,6 +53,8 @@ service PD {
 
     rpc ScanRegions(ScanRegionsRequest) returns (ScanRegionsResponse) {}
 
+    rpc BatchScanRegions(BatchScanRegionsRequest) returns (BatchScanRegionsResponse) {}
```
Comment (rleungx) [inline proto/pdpb.proto:56]: Will it replace ScanRegions?

Comment (you06) [reply proto/pdpb.proto:56]: Yes, I think `ScanRegions` can be deprecated then.

Comment (rleungx) [reply proto/pdpb.proto:56]: Then how about adding a comment for it?

Comment (you06) [reply proto/pdpb.proto:57]: Done.

**Review by rleungx — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Pull request merged**